docs/networking/split-horizon_everest_operator_changes.md

```diff
@@ -0,0 +1,148 @@
+|--------------------------------------|-----------------------------------------------------------------------------|
+| `.spec.baseDomainNameSuffix`          | Domain name suffix used for generating full Pod hostnames in the Replica Set. |
+| `.spec.tls.secretName`                | Name of the Kubernetes secret that contains the TLS certificate.             |
+| `.spec.tls.certificate.certFile`      | Base64-encoded TLS certificate file (`.crt`).                                |
+| `.spec.tls.certificate.keyFile`       | Base64-encoded TLS private key file.                                         |
+| `.spec.tls.certificate.caCertFile`    | Base64-encoded CA certificate file (`ca.crt`).                               |
+
+## Allowed TLS configuration options
+
+When creating the Custom Resource (CR), provide one of the following combinations:
+
+- `.spec.tls.secretName` only – Use this option if a TLS certificate already exists and should be reused.
+
+- `.spec.tls.secretName + .spec.tls.certificate.*` – In this case, the values from `.spec.tls.certificate.*` will be copied into a newly created Secret named `.spec.tls.secretName.` 
+
+    Afterwards, the `.spec.tls.certificate.*` values will be removed for security reasons.
+
+!!! note
+    Percona Everest does not validate whether the provided TLS certificate matches the specified base domain name.
+
+## Split-Horizon DNS configuration handling in Percona Everest Operator
+
+### SplitHorizonDNSConfig CR handling
+
+Once a new `SplitHorizonDNSConfig` object is created, the Percona Everest operator will handle it as follows:
+{.power-number}
+
+1. Validation
+    - Validate that the field `.spec.baseDomainNameSuffix' contains a valid domain name.
+    - Verify that `.spec.tls.secretName` is provided. This can either reference an existing secret or will be used to create a new one.
+
+2. TLS secret management
+
+    - If .spec.tls.certificate.* is not provided, ensure a secret with name `.spec.tls.secretName` exists in the same namespace and is of type TLS.
+
+    - If `.spec.tls.certificate.*` is provided, create a secret named `.spec.tls.secretName` if it does not already exist, and copy the certificate values from `.spec.tls.certificate.*` into that secret.
+
+
+### DatabaseCluster CR extension
+
+The `DatabaseCluster` CR is extended with an optional field `.spec.engineFeatures.psmdb.split-horizon-dns`.
+
+The default value is **nil**, indicating that the Split-Horizon DNS feature will not be applied to the Percona Operator for the MongoDB cluster.
+
+
+??? example "Example"
+    ```yaml
+    apiVersion: everest.percona.com/v1alpha1
+    kind: DatabaseCluster
+    metadata:
+        name: psmdb-4d5
+        namespace: ns-1
+    spec:
+        engine:
+            type: psmdb
+            userSecretsName: everest-secrets-psmdb-4d5
+        engineFeatures:
+            psmdb:
+                splitHorizonDnsConfigName: mycompany.com
```

Comment:
check carefully with the spec: https://www.notion.so/percona/Split-Horizon-DNS-feature-for-PSMDB-27f674d091f3806b9d46dffaa7e24866?source=copy_link#280674d091f380fb8c65f91ecfe96d53